src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -330,6 +417,258 @@ public void jobFinished () {
+        lines.add(String.format("LOGFILE=/var/log/%s.log", tripPlanner));
+        // Log user data setup to /var/log/user-data.log
+        lines.add("exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1");
+        // Create the directory for the graph inputs.
+        lines.add(String.format("mkdir -p %s", routerDir));
+        lines.add(String.format("chown ubuntu %s", routerDir));
+        // Remove the current inputs and replace with inputs from S3.
+        lines.add(String.format("rm -rf %s/*", routerDir));
+        lines.add(String.format("aws s3 --region us-east-1 cp %s /tmp/bundle.zip", s3BundlePath));
+        lines.add(String.format("unzip /tmp/bundle.zip -d %s", routerDir));
+        // FIXME: Add ability to fetch custom bikeshare.xml file (CarFreeAtoZ)
+        if (false) {
+            lines.add(String.format("wget -O %s/bikeshare.xml ${config.bikeshareFeed}", routerDir));
+            lines.add(String.format("printf \"{\\n  bikeRentalFile: \"bikeshare.xml\"\\n}\" >> %s/build-config.json\"", routerDir));
+        }
+        // Download trip planner JAR.
+        lines.add(String.format("mkdir -p %s", jarDir));
+        lines.add(String.format("wget %s -O %s/%s.jar", s3JarUrl, jarDir, jarName));
+        if (graphAlreadyBuilt) {
+            lines.add("echo 'downloading graph from s3'");
+            // Download Graph from S3 and spin up trip planner.
+            lines.add(String.format("aws s3 --region us-east-1 cp %s %s/Graph.obj", getS3GraphPath(), routerDir));
+        } else {
+            lines.add("echo 'starting graph build'");
+            // Build the graph if Graph object (presumably this is the first instance to be started up).
+            if (deployment.r5) lines.add(String.format("sudo -H -u ubuntu java -Xmx6G -jar %s/%s.jar point --build %s", jarDir, jarName, routerDir));
+            else lines.add(String.format("sudo -H -u ubuntu java -jar %s/%s.jar --build %s > $BUILDLOGFILE 2>&1", jarDir, jarName, routerDir));
+            // Upload the graph to S3.
+            if (!deployment.r5) lines.add(String.format("aws s3 --region us-east-1 cp %s/Graph.obj %s", routerDir, getS3GraphPath()));
+        }
+        if (deployment.buildGraphOnly) {
+            lines.add("echo 'shutting down server (build graph only specified in deployment target)'");
+            lines.add("sudo poweroff");
+        } else {
+            lines.add("echo 'kicking off trip planner (logs at $LOGFILE)'");
+            // Kick off the application.
+            if (deployment.r5) lines.add(String.format("sudo -H -u ubuntu nohup java -Xmx6G -Djava.util.Arrays.useLegacyMergeSort=true -jar %s/%s.jar point --isochrones %s > /var/log/r5.out 2>&1&", jarDir, jarName, routerDir));
+            else lines.add(String.format("sudo -H -u ubuntu nohup java -jar %s/%s.jar --server --bindAddress 127.0.0.1 --router default > $LOGFILE 2>&1 &", jarDir, jarName));
+        }
+        return String.join("\n", lines);
```

Comment:
Basically all of these statements are susceptible to failure and don't have an easier way to recover. Also, there is no real way to check on the progress of everything. Maybe this script could also be updating some file that could be on s3 that datatools-server could be polling to see how much progress has been made and whether or not a failure occurred. Otherwise, if an error occurs the user won't know what was the issue and will have to wait an hour for the job to be failed.